src/core/registration.c

```diff
@@ -247,6 +249,18 @@ MsQuicRegistrationShutdown(
             Entry = Entry->Flink;
         }
 
+        CxPlatLockAcquire(&Registration->ListenerLock);
+        Entry = Registration->Listeners.Flink;
+        while (Entry != &Registration->Listeners) {
+
+            QUIC_LISTENER* Listener =
+                CXPLAT_CONTAINING_RECORD(Entry, QUIC_LISTENER, RegistrationLink);
+
+            Entry = Entry->Flink;
+            MsQuicListenerStop((HQUIC)Listener);
```

Comment:
We can't do this under lock. Instead, move `Registration->Listeners` into a local list while under lock. Then, outside the lock, do this work.

Reply:
ok. I will do the work after releasing the lock.

We don't need a local list because after  `Registration->ShuttingDown = TRUE;` , neither `MsQuicListenerOpen` nor `QuicListenerFree` can access/update this list since I added a check there.

do you agree?

Reply:
Yeah, I think that's fine. Thanks!